MonsterTrainModdingAPI/Managers/CustomAssetManager.cs

```diff
@@ -96,18 +106,179 @@ public static void UnloadAssetBundle(string bundleName, bool unloadAllLoadedObje
             }
         }
 
+
         /// <summary>
         /// Helper class optionally used to aid in loading asset bundles.
         /// </summary>
         public class AssetBundleLoadingInfo
         {
+            public IDictionary<Type, ISettings> LoadingDictionary { get; private set; } = new Dictionary<Type, ISettings>();
             public string AssetName { get; set; }
             public string BundlePath { get; set; }
-            public AssetBundleLoadingInfo(string assetName, string bundlePath = "")
+            public AssetBundleLoadingInfo(string assetName, string bundlePath)
             {
                 this.AssetName = assetName;
                 this.BundlePath = bundlePath;
             }
+
+            public AssetBundleLoadingInfo AddImportSettings<T>(T ImportSettings) where T : ISettings
+            {
+                foreach(Type inter in ImportSettings.GetType().GetInterfaces())
+                {
+                    if (inter.IsGenericType && (typeof(ISettings).IsAssignableFrom(inter)))
+                    {
+                        Type type = inter.GetGenericArguments()[0];
+                        LoadingDictionary.Add(type, ImportSettings);
+                        return this;
+                    }
+                }
+                throw new ArgumentException("ISettings<Type> was not found to be implimented in " + ImportSettings.GetType());
+            }
+        }
+
+        /// <summary>
+        /// An Interface used to Represent Settings
+        /// </summary>
+        public interface ISettings { }
```

Comment:
Is this interface necessary? Serious question. I'm not well-versed in generic interfaces.